src/electionguard/group.cpp

```diff
@@ -755,38 +755,32 @@ namespace electionguard
     unique_ptr<ElementModQ> a_plus_bc_mod_q(const ElementModQ &a, const ElementModQ &b,
                                             const ElementModQ &c)
     {
-        // since the result of b*c can be larger than Q's 256,
-        // use P's length and 4096 Hacl for rest of calculation
+        // multiply b * c and the result will be twice Q in size
+        uint64_t bc[MAX_Q_LEN * 2] = {};
+        Hacl_Bignum256_mul(const_cast<ElementModQ &>(b).get(), const_cast<ElementModQ &>(c).get(),
+                           bc);
 
-        uint64_t resultBC[MAX_P_LEN] = {}; // init to MAX_P_LEN
-        Bignum256::mul(const_cast<ElementModQ &>(b).get(), const_cast<ElementModQ &>(c).get(),
-                       static_cast<uint64_t *>(resultBC));
-
-        auto bc_as_p = make_unique<ElementModP>(resultBC, true);
-        auto a_as_p = a.toElementModP();
+        // perform the mod operation on bc
+        uint64_t bc_mod_q[MAX_Q_LEN] = {};
+        const auto &q = Q();
+        bool modSuccess = Hacl_Bignum256_mod(q.get(), bc, bc_mod_q);
+        if (!modSuccess) {
+            throw runtime_error("a_plus_bc_mod_q mod operation failed");
+        }
 
-        uint64_t a_plus_bc_result[MAX_P_LEN_DOUBLE] = {};
+        uint64_t a_plus_bc[MAX_Q_LEN * 2] = {};
```

Comment:
If one does not already exist, we should use/make a MAX_Q_LEN_DOUBLE like we already have for MAX_P_LEN_DOUBLE instead.

Reply:
Sounds good, change made.